iocore/eventsystem/P_UnixSocketManager.h

```diff
@@ -55,21 +55,52 @@ transient_error()
   return transient;
 }
 
+#if !HAVE_ACCEPT4
+int
+accept4(int sockfd, struct sockaddr *addr, socklen_t *addrlen, int flags)
+{
+  int fd, err;
```

Comment:
Unfortunately ``ats_scoped_fd`` won't preserve errno. The ``SocketManager`` routines are inconsistent about whether they return the error or set ``errno``. If you can clean than up (separately?) then ``ats_scoped_fd`` is a good option.